stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -184,6 +184,50 @@ function eigen!(A::AbstractMatrix, C::Cholesky; sortby::Union{Function,Nothing}=
     GeneralizedEigen(sorteig!(vals, vecs, sortby)...)
 end
 
+# Bunch-Kaufmann (LDLT) based solution for generalized eigenvalues and eigenvectors
+function eigen(A::AbstractMatrix{T}, B::BunchKaufman{T,<:StridedMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    eigen!(copy(A), copy(B); sortby)
```

Comment:
Is your algorithm mutating `B`? If not, then you don't need the copy, and the added function in `bunchkaufman.jl`.

Reply:
Unfortunately, a call to `syconv!` or `syconvf_rook!` is in-place and modifies `B.LD`. The usual `getproperty(B, :L)` creates a copy. Hence, the copy `copy(B)` is necessary.

Reply:
Okay, but if both calls make a copy, then why not leave it to `getproperty`?